README.md

```diff
@@ -129,77 +100,113 @@ For some large and heavily trafficked sites, that case is rare. For others, it
+    * The dictionary resource will be downloaded with CORS in “omit” mode to discourage including user-specific private data in the dictionary.
+    * It will be downloaded with “idle” priority, once the site is actually idle.
     * Browsers may decide to not download it when they suspect that the user is paying for bandwidth, or when used by sites that are not likely to amortize the dictionary costs (e.g. sites that the user isn’t visiting frequently enough).
     * Browsers may decide to not use a shared dictionary if it contains hints that its contents are not public (e.g. `Cache-Control: private` headers).
-* Once the dictionary is downloaded, future navigation requests may advertise the presence of the dictionary using a `sec-bikeshed-available-dictionary:` request header.
-    * Only a single shared dictionary will be advertised. Shared dictionary will only be used if the path in question doesn't have another available dictionary (from the static resource flow). New shared dictionaries will override older ones.
-    * Browsers may avoid advertising the header in some cases to reduce abuse risk.
-    * Due to the dynamic nature of the resource, dynamically compressing it with the dictionary will not pose a regression compared to common practices today (e.g. serving-time compression with brotli level 5).
-    * Similar to the static resource case, the server will indicate a use of the dictionary using the `content-encoding` header.
-    * Similarly again, in case the browser advertized a dictionary but then fails to successfuly fetch it from its cache *and* the dictionary was used by the server, the resource request should be terminated.
+* The dictionary response must include the `bikeshed-use-as-dictionary: <path>` header, appropriate cache lifetime headers and will be used for future requests using the same process as the [Static resources flow](#static-resources-flow).
+
+### Compression API
 
-#### Open questions:
-* Should shared dictionaries also have a "path" component, enabling multiple ones for different parts of the origin?
+The compression API can also expose support for using caller-supplied dictionaries but that is out-of-scope for this proposal.
 
+### Security mitigations
 
-### Compression API
+Since the contents of the dictionary and compressed resource are both effectively readable through side-channel attacks, this proposal makes it excplicit and requires that both be CORS-readable from the document origin. The dictionary and compressed resource must also be from the same origin as each other with the `scope` only comprising the path component of the matching URL.
 
-The above can also be exposed to the Compression API.
+For dictionaries and resources that are same-origin as the document, no additional requirements exist as both are CORS-readable from the document context.
```

Comment:
Should we mention navigation requests explicitly? (as same-origin)

Reply:
Added. Let me know if I was too wordy.